wurst/systems/entities/bosses/AlligatorMan.wurst

```diff
@@ -26,6 +49,42 @@ public class AlligatorMan extends Hostile
     override function getNumCorpses() returns int
         return 10
 
-
 init
     AlligatorMan.initialize()
+
+    //Cycle through alligator regions
+    for curr_region in REGIONS
+        //Create trigger for detecting alligator
+        let x =CreateTrigger()
+        x.registerEnterRegion(curr_region, null)
+        x.addAction()->
+            let trigger_region = GetTriggeringRegion()
+            let entering_unit = GetEnteringUnit()
+            //Verify entering unit is alligator
+            if entering_unit.getTypeId() == UNIT_ALIGATOR_MAN
+                //Check to make sure we don't already have that alligator
+                if not regionsToAlligators.has(trigger_region)
```

Comment:
You don't need this check because you destroy the trigger before this ever evaluates to false.